### PR TITLE
chore: update data root tuple genesis comment

### PR DESCRIPTION
**Pull request by SweeXordious**
## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Review by evan-forbes — APPROVED** [review]

File: src/DataRootTuple.sol [hunk 1]
```diff
@@ -7,7 +7,7 @@ pragma solidity ^0.8.4;
 ///  https://github.com/celestiaorg/celestia-specs/blob/master/src/specs/data_structures.md#header
 struct DataRootTuple {
     // Celestia block height the data root was included in.
-    // Genesis block is height = 1.
+    // Genesis block is height = 0.
```
Comment (rootulp) [inline src/DataRootTuple.sol:10]: [question] why `0`? Blockspace race started at block height `1` per https://github.com/celestiaorg/networks/blob/309f961dcdc7e7353e8143cc59c1797e4c0fc326/blockspacerace/genesis.json#L4

Comment (SweeXordious) [reply src/DataRootTuple.sol:10]: I guess the initial height specified the first block, not the genesis. When I run a local celestia-app:
```
1:23AM INF starting node with ABCI Tendermint in-process
1:23AM INF service start impl=multiAppConn module=proxy msg={}
1:23AM INF service start connection=query impl=localClient module=abci-client msg={}
1:23AM INF service start connection=snapshot impl=localClient module=abci-client msg={}
1:23AM INF service start connection=mempool impl=localClient module=abci-client msg={}
1:23AM INF service start connection=consensus impl=localClient module=abci-client msg={}
1:23AM INF service start impl=EventBus module=events msg={}
1:23AM INF service start impl=PubSub module=pubsub msg={}
1:23AM INF service start impl=IndexerService module=txindex msg={}
1:23AM INF ABCI Handshake App Info hash= height=0 module=consensus protocol-version=0 software-version=0.14.0-rc0-11-ge905143e8
1:23AM INF ABCI Replay Blocks appHeight=0 module=consensus stateHeight=0 storeHeight=0
1:23AM INF initializing blockchain state from genesis.json
1:23AM INF asserting crisis invariants inv=1/11 module=x/crisis name=bank/nonnegative-outstanding
1:23AM INF asserting crisis invariants inv=2/11 module=x/crisis name=bank/total-supply
1:23AM INF asserting crisis invariants inv=3/11 module=x/crisis name=staking/module-accounts
1:23AM INF asserting crisis invariants inv=4/11 module=x/crisis name=staking/nonnegative-power
1:23AM INF asserting crisis invariants inv=5/11 module=x/crisis name=staking/positive-delegation
1:23AM INF asserting crisis invariants inv=6/11 module=x/crisis name=staking/delegator-shares
1:23AM INF asserting crisis invariants inv=7/11 module=x/crisis name=distribution/nonnegative-outstanding
1:23AM INF asserting crisis invariants inv=8/11 module=x/crisis name=distribution/can-withdraw
1:23AM INF asserting crisis invariants inv=9/11 module=x/crisis name=distribution/reference-count
1:23AM INF asserting crisis invariants inv=10/11 module=x/crisis name=distribution/module-account
1:23AM INF asserting crisis invariants inv=11/11 module=x/crisis name=gov/module-account
1:23AM INF asserted all invariants duration=1.170572 height=0 module=x/crisis
1:23AM INF created new capability module=ibc name=ports/transfer
1:23AM INF port binded module=x/ibc/port port=transfer
1:23AM INF claimed capability capability=1 module=transfer name=ports/transfer
1:23AM INF Completed ABCI Handshake - CometBFT and App are synced appHash= appHeight=0 module=consensus
1:23AM INF Version info abci=0.17.0 block=11 cmtbft_version=0.34.27 commit_hash= p2p=8
1:23AM INF This node is a validator addr=BF55887140F913AB1E19A31661095178583F1D2B module=consensus pubKey=sTTWK3RO82LzkPTTNNVEWPktaFBbeoikqMCYqiZyp2s=
1:23AM INF P2P Node ID ID=dc328c054debe7c4987a3501df02a378858b482d file=/home/midnight/.celestia-app/config/node_key.json module=p2p
1:23AM INF Adding persistent peers addrs=[] module=p2p
1:23AM INF Adding unconditional peer ids ids=[] module=p2p
1:23AM INF Add our address to book addr={"id":"dc328c054debe7c4987a3501df02a378858b482d","ip":"0.0.0.0","port":26656} book=/home/midnight/.celestia-app/config/addrbook.json module=p2p
1:23AM INF service start impl=Node msg={}
1:23AM INF Starting pprof server laddr=localhost:6060
1:23AM INF serve module=rpc-server msg={}
1:23AM INF service start impl="P2P Switch" module=p2p msg={}
1:23AM INF service start impl=ConsensusReactor module=consensus msg={}
1:23AM INF Reactor  module=consensus waitSync=false
1:23AM INF service start impl=ConsensusState module=consensus msg={}
1:23AM INF service start impl=baseWAL module=consensus msg={} wal=/home/midnight/.celestia-app/data/cs.wal/wal
1:23AM INF service start impl=Group module=consensus msg={} wal=/home/midnight/.celestia-app/data/cs.wal/wal
1:23AM INF service start impl=TimeoutTicker module=consensus msg={}
1:23AM INF Searching for height height=1 max=0 min=0 module=consensus wal=/home/midnight/.celestia-app/data/cs.wal/wal
1:23AM INF Searching for height height=0 max=0 min=0 module=consensus wal=/home/midnight/.celestia-app/data/cs.wal/wal
1:23AM INF Found height=0 index=0 module=consensus wal=/home/midnight/.celestia-app/data/cs.wal/wal
1:23AM INF Catchup by replaying consensus messages height=1 module=consensus
1:23AM INF Replay: Done module=consensus
1:23AM INF service start impl=Evidence module=evidence msg={}
1:23AM INF service start impl=StateSync module=statesync msg={}
1:23AM INF service start impl=PEX module=pex msg={}
1:23AM INF service start book=/home/midnight/.celestia-app/config/addrbook.json impl=AddrBook module=p2p msg={}
1:23AM INF service start impl=Mempool msg={}
1:23AM INF service start impl=BlockchainReactor module=blockchain msg={}
1:23AM INF Saving AddrBook to file book=/home/midnight/.celestia-app/config/addrbook.json module=p2p size=0
1:23AM INF Ensure peers module=pex numDialing=0 numInPeers=0 numOutPeers=0 numToDial=10
1:23AM INF No addresses to dial. Falling back to seeds module=pex
1:23AM INF starting API server... module=api-server
1:23AM INF serve module=api-server msg={}
1:23AM INF Timed out dur=6.624471 height=1 module=consensus round=0 step=1
1:23AM INF received proposal module=consensus proposal={"Type":32,"block_id":{"hash":"3CDD995CE3288B15B6FF2F06258B0B6111DBA6626F3DFEEDAD0E6449F3750D45","parts":{"hash":"A58C03A94EE532C4F3286B7DAC39FD6AC4448FFBE77E217370DCB5A3BF7C950F","total":1}},"height":1,"pol_round":-1,"round":0,"signature":"pzA7/R2pa1yMMQa2Ox5tm5KtoT/EqPdcQHx94qIXffAD4joGgCbQd11YHU4vXHhlY3drmpIXnxFvY4z9gTgiBw==","timestamp":"2023-04-17T23:23:16.627083375Z"}
1:23AM INF received complete proposal block hash=3CDD995CE3288B15B6FF2F06258B0B6111DBA6626F3DFEEDAD0E6449F3750D45 height=1 module=consensus
1:23AM INF finalizing commit of block hash={} height=1 module=consensus num_txs=0 root=E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855
1:23AM INF minted coins from module account amount=20597260utia from=mint module=x/bank
1:23AM INF executed block height=1 module=state num_invalid_txs=0 num_valid_txs=0
1:23AM INF commit synced commit=436F6D6D697449447B5B32333920383520313432203733203130372035302038372031333620333520313039203139362037332031313620323434203838203330203136342036382032333620333120313130203437203330203531203234362031323720323133203135372031313520313130203133372037385D3A317D
1:23AM INF committed state app_hash=EF558E496B325788236DC44974F4581EA444EC1F6E2F1E33F67FD59D736E894E height=1 module=state num_txs=0
1:23AM INF indexed block exents height=1 module=txindex
```
The logs state `height=0` as an existing height (genesis),  then starts blocks at `height=1`.

This parameter was added here https://github.com/tendermint/tendermint/pull/5191, which probably means that genesis height will be `initial_height-1`.
Maybe @cmwaters has more information.

Comment (evan-forbes) [reply src/DataRootTuple.sol:10]: I believe heights start at 1, and 0 is considered the genesis block.

when querying for block at height 0, we get
```json
{
  "jsonrpc": "2.0",
  "id": -1,
  "error": {
    "code": -32603,
    "message": "Internal error",
    "data": "height must be greater than 0, but got 0"
  }
}
```

**Review by evan-forbes — APPROVED** [review]

File: src/DataRootTuple.sol (hunk 1 above)
Comment (adlerjohn) [reply src/DataRootTuple.sol:10]: In that case, the comment should be updated to "first queryable block is that height 1"?

Comment (SweeXordious) [reply src/DataRootTuple.sol:10]:
```suggestion
    // first queryable block is height = 1.
```

Comment (adlerjohn) [reply src/DataRootTuple.sol:10]: Yeah. You could keep the comment on the genesis block being at height 0 too.

**Review by adlerjohn — APPROVED** [review]

**Review by liamsi — APPROVED** [review]

File: src/DataRootTuple.sol [hunk 2]
```diff
@@ -7,7 +7,8 @@ pragma solidity ^0.8.4;
...
```
Comment (liamsi) [inline src/DataRootTuple.sol:10]: One minor nit: It is the genesis file that has height 0. The block at height 1 will reference the genesis file as its parent. It is only semantics but I would argue that the first _block_ is the one produced by consensus and the genesis file does not really count as a block as it is not produced by the same consensus protocol. 

Question: Does the genesis file / the initial chain state play any role here at all? Isn't the first relevant block only that with height 1? If not, can you clarify how the genesis file plays a role in this smart contract?

Comment (SweeXordious) [reply src/DataRootTuple.sol:10]: It doesn't play any role in the smart contract, it's just a comment to specify from where the data commitments will start.

**Pull request merged**